search-app/src/cmr/search/site/data.clj

```diff
@@ -99,7 +112,9 @@
         result (query-exec/execute-query context query)
         {:keys [items granule-counts-map]} result]
     (sort-by :entry-title
-             (map #(assoc % :granule-count (get granule-counts-map (:concept-id %))) items))))
+            (map #(assoc % :granule-count 
```

Comment:
Although this code works (It processes one collection at a time to get the granule counts), it is not efficient. As we have talked, you should execute an aggregation search to get the matching granule count for all collections similar to how `cmr.search.services.query-execution.granule-counts-results-feature` does it.

Reply:
done